assets/js/theme/common/carousel/utils/handleImageAspectRatio.js

```diff
@@ -32,15 +40,11 @@ export default ({ delegateTarget }, carousel) => {
 
     if ($activeSlide.find('.heroCarousel-content').length) return;
 
-    $('<img />')
-        .on('load', function getImageSizes() {
-            const imageHeight = this.height;
-            const imageWidth = this.width;
-
-            if (imageHeight < 2 || imageWidth < 2) return;
-
-            const imageAspectRatio = imageHeight / imageWidth;
-            $activeSlideAndClones.each((idx, slide) => $(slide).addClass(defineClass(imageAspectRatio)));
-        })
-        .attr(attrsObj);
+    if (activeSlideImgNode.complete) {
+        if (activeSlideImgNode.naturalHeight === 1) {
+            $activeSlideAndClones.each((idx, slide) => $(slide).data(IS_ANALYZED_DATA_ATTR, false));
+        } else if (activeSlideImgNode.naturalHeight > 1) {
+            installClass(activeSlideImgNode, $activeSlideAndClones);
+        }
+    } else $activeSlideImg.on('load', () => installClass(activeSlideImgNode, $activeSlideAndClones));
```

Comment:
Should we add error style class if load was failed?

Reply:
we should not, because handleImageLoad util is responsible for it